data/mods/ssb/abilities.ts

```diff
@@ -232,6 +232,31 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
 		},
 	},
 
+	// Ganjafin
+	gamblingaddiction: {
+		shortDesc: "When under 1/4 max HP, +1 Spe, heal to full HP, and all moves become Final Gambit.",
+		name: "Gambling Addiction",
+		onResidualOrder: 29,
+		onResidual(pokemon) {
+			if (pokemon.hp && pokemon.hp < pokemon.maxhp / 4) {
```

Comment:
you need a second check, or else somehow this mon is gonna keep recovering back to full